docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+   tenants in the cluster via RBAC rules, it's not possible to prevent them from
+   _using_ those credentials solely with RBAC.
+
+As discussed in detail in the [Crossplane Multi-tenancy Guide][xp-mt],
+Crossplane is opinionated about the different personas in an organization adopting
+Crossplane. We make a distinction between the _infrastructure operators_ (or
+_platform builders_) who are expected to manage cluster-scoped resources (like
+`ProviderConfig`s, XRDs and `Composition`s) and _application operators_, who are
+expected to consume the infrastructure for their applications. And tenant
+subjects are classified as _application operators_, i.e., it's the
+infrastructure operator's responsibility to manage the infrastructure _across_
+the tenants via cluster-scoped Crossplane resources, and it's possible and
+desirable from an isolation perspective to disallow application operators, who
+are tenant subjects, to directly access these shared cluster-scoped resources.
+This distinction is currently possible with Crossplane because: 
+1. Crossplane `Claim` types are defined via cluster-scoped XRDs by
+   infrastructure operators and _namespaced_ `Claim` instances are used by the
+   tenant subjects. This allows infrastructure operators to define RBAC rules
+   that allow tenant subjects to only access resources in their respective
+   namespaces, e.g., `Claim`s.
+1. However, [1] is not sufficient on itself, as the scheme is still prone to
+   privilege escalation attacks if the API exposed by the XR is not well designed. The
+   (shared) provider `ServiceAccount` has access to all Cloud provider
+   credentials in the cluster and if the exposed XR API allows a `Claim` to
+   reference cross-tenant `ProviderConfig`s, then a misbehaving tenant subject
+   can `create` a `Claim` which references some other tenant's credential set.
+   Thus in our multi-tenancy [guide][xp-mt], we propose a security scheme where:
+    1. The infrastructure operator follows a specific naming convention for the
+       `ProviderConfig`s she provisions: The `ProviderConfig`s for different
+       tenants are named after those tenants' namespaces. 
+    2. The infrastructure operator carefully designs `Composition`s that patch
+       `spec.providerConfigRef` of composed resources using the `Claim`'s
+       namespace. 
+    3. Tenant subjects are **not** allowed to provision managed resources directly (and also
+       XRDs or `Composition`s) but only `Claim`s in their namespaces. And any
+       `Composition` they can select with their `Claim`s will compose resources
+       that refer to a `ProviderConfig` provisioned for their tenant (the
+       `ProviderConfig` with the same name as the tenant's namespace). 
+    4. We also suggest that the naming conventions imposed by this scheme on
+       `ProviderConfig`s can be relaxed to some degree by using `Composition`'s
+       [patching capabilities][ref-compositions]. For instance, a string
```

Comment:
The `[patching capabilities]` link is broken.

Reply:
Unfortunately can't find document link to resolve this.

Reply:
Proposed a fix for the reference `ref-compositions` below.

Reply:
Fixed